dashboard/app/models/school_info.rb

```diff
@@ -133,13 +134,14 @@ def sync_from_schools
   validate :validate_with_country
   validate :validate_without_country
   validate :validate_zip
+  validate :validate_complete
 
   def usa?
     ['US', 'USA', 'United States'].include? country
   end
 
-  def should_validate?
-    validation_type != VALIDATION_NONE
+  def should_check_for_full_or_no_validation?
```

Comment:
nit: I'd probably prefer `should_check_for_full_validation` (I'm not sure what the `or_no` bit is referring to) but it's up to you.